### DISCUSSION TITLE
PB as Framework (JS) record.load doesn't work with e.model.get("data")

### DISCUSSION THREAD

**Discussion by ericbaek** [pocketbase/pocketbase#4385] [Q&A] [answered]
![20240220 - ticket pb js — Pocketbase 0 21 3 ARM64 (Code) 2@2x](https://github.com/pocketbase/pocketbase/assets/43541203/78fb2d23-556a-416b-b1ea-d02dabb17ed8)

I am not sure if this bug or I am doing something wrong. Using `onModelBeforeCreate` to make framework JS hooks.

This is what console.log prints out.
![20240220 - ticket pb js — Pocketbase 0 21 3 ARM64 (Code) 3@2x](https://github.com/pocketbase/pocketbase/assets/43541203/9ce1928b-ad85-440f-b789-0719c8b1f8ed)

However, as you see Record is saved with null. I tried putting actual data within the code and it works fine. So I doubt data is wrong.
![20240220 - arcadeBizhour - Acme - PocketBase (Arc)@2x](https://github.com/pocketbase/pocketbase/assets/43541203/c285d1bd-091e-45be-91de-872d7b9cf17b)

**Comment by ganigeorgiev** [comment, marked as answer]
Please don't post code as screenshots. It is difficult to read and I have no easy way to copy your example to try to reproduce it.

If `data` in the above is coming from a `json` type field then most likely it doesn't work because the value is not a plain object but `types.JsonRaw`, aka. you may need to wrap the `e.model.get("data")` with `JSON.parse(...)`.

In any case, testing with the below code locally works fine for me:

```js
const collection = $app.dao().findCollectionByNameOrId("example")

const record = new Record(collection)

record.load({
    "a": "test",
    "b": {"something1": 123},
    "c": {"something2": [1, 2, 3]},
})

$app.dao().saveRecord(record)
```

**Reply by ericbaek** [reply]
`Json.parse()` solved the issue. Thanks
Also, sorry for the screenshot. I will make sure to paste the code block next time. 😉
